gensim/models/keyedvectors.py

```diff
@@ -916,11 +915,11 @@ def wmdistance(self, document1, document2):
                 if t2 not in docset2 or distance_matrix[i, j] != 0.0:
                     continue
 
-                # Compute Euclidean distance between unit-normed word vectors.
+                # Compute Euclidean distance between (potentially unit-normed) word vectors.
                 distance_matrix[i, j] = distance_matrix[j, i] = np.sqrt(
-                    np_sum((self.get_vector(t1, norm=True) - self.get_vector(t2, norm=True))**2))
+                    np_sum((self.get_vector(t1, norm=norm) - self.get_vector(t2, norm=norm))**2))
 
-        if np_sum(distance_matrix) == 0.0:
+        if abs(np_sum(distance_matrix)) < 1e-8:
```

Comment:
This looks like a fix or partial fix for a different issue. I'm not sure, but if all pairwise vocab distances are 0.0 (or sufficiently close?), isn't a return value of 0.0 more likely to match intuitions about what the distance between "essentially identical" texts should be?

And, if there is some actual range of tiny-but-not-technically-zero distances that either (a) confuse  EMD/PyEMD; or (b) justify clipping-the-result-to-zero, are we sure `1e-8` is the well-grounded threshold, based on some reasoning about the source of such noise? Or should the threshold be highlighted as something more tentative, or even configurable?

Reply:
(Separate thought: if it is only true `0.0` that causes `emd` problems, maybe the better solution is to replace zeros with tiny noise nonzero values?)

Reply:
This wasn't a fix for any particular issue. Just when I visited the code now, I saw a bunch of weird constructs there – this one of them (comparing a float for exact equality, bad smell).

I'm not aware of any related user report or bugs.

Reply:
Aha - my unconfirmed suspicion had been one of the other recent WMD problem reports might have been due to unexpected/excessive 0-distances, so changes here seemed potentially related. Still, seems odd that zeros should generate a loud warning and max (`inf`) distance, rather than just `0.0`. (It looks to me like this code might report `inf` as the WMD between `['foo', 'bar']` & `['bar', 'foo']` instead of `0.0`.) And, I'd be reluctant to change an absolute-equality test to a less-than-some-threshold test unless I knew for sure a test case where that was required.

Reply:
Yeah, I'm not worried treating distance matrices whose total sum is <0.00000001 as 0.0.

If there's a bug with infinite WMD distances let's treat that separately. This change really is unrelated.

Reply:
But: you've changed the behavior for tiny-distances returning a tiny-WMD to instead returning `inf` - without any illustrative test cases. (That looks to me like it took an existing non-intuitive behavior/error in the code and expanded it to apply to **more** cases.) On a simple, "if it ain't broken don't fix it" analysis, why take that risk?

Reply:
The previous behaviour was broken – comparing a sum of many floats against a fixed value cannot be right.

I agree the discontinuity around `inf` is weird. The algo, and its edge cases, deserve a thorough check. Can you investigate?